app/services/trainees/filter.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Trainees
+  class Filter
+    def self.call(trainees:, filters:)
```

Comment:
I think we could use our service pattern here for consistency. Have a look at one of the other services but we can treat this as an instance that would be created but wrapped in a Class factory method to instantiate. It'll also means you can keep some of the state in the instance and save on each method requiring an argument since they're all private to it anyway.

Reply:
good idea!